dipy/core/sphere.py

```diff
@@ -498,6 +500,155 @@ def disperse_charges(hemi, iters, const=.2):
     return HemiSphere(xyz=charges), potential
 
 
+def _equality_constraints(vects, *args):
+    """Spherical equality constraint. Returns 0 if vects lies on the unit
+    sphere.
+
+    Parameters
+    ----------
+    vects : array-like shape (N * 3)
+        Points on the sphere.
+    Returns
+    -------
+    array-like (N,)
+        Difference between squared vector norms and 1.
+
+    """
+
+    N = vects.shape[0] / 3
+    vects = vects.reshape((N, 3))
+    return (vects ** 2).sum(1) - 1.0
+
+
+def _grad_equality_constraints(vects):
+    """Return normals to the surface constraint (which corresponds to
+    the gradient of the implicit function).
+
+    Parameters
+    ----------
+    vects : array-like (N * 3)
+        Points on the sphere.
+    Returns
+    -------
+    array-like (N, N * 3)
+        grad[i, j] contains :math:`\partial f_i / \partial x_j`.
+
+    """
+
+    N = vects.shape[0] / 3
+    vects = vects.reshape((N, 3))
+    vects = (vects.T / np.sqrt((vects ** 2).sum(1))).T
+    grad = np.zeros((N, N * 3))
+    for i in range(3):
+        grad[:, i * N:(i + 1) * N] = np.diag(vects[:, i])
+    return grad
+
+
+def _get_forces_alt(vects, alpha=2.0, **kwargs):
+    """Electrostatic-repulsion objective function. The alpha parameter
+    controls the power repulsion (energy varies as $1 / r^\alpha$).
```

Comment:
What about saying something like "For `alpha = 1.0`, this corresponds to electrostatic interaction energy."?

Reply:
Addressed in b693c60.